Neos.EventSourcedContentRepository/Classes/Domain/Context/NodeAggregate/Feature/NodeMove.php

```diff
@@ -96,12 +96,14 @@ public function handleMoveNodeAggregate(MoveNodeAggregate $command): CommandResu
                     [$command->getNewParentNodeAggregateIdentifier()]
                 );
 
-                $this->requireNodeNameToBeUncovered(
-                    $command->getContentStreamIdentifier(),
-                    $nodeAggregate->getNodeName(),
-                    $command->getNewParentNodeAggregateIdentifier(),
-                    $affectedDimensionSpacePoints
-                );
+                if ($nodeAggregate->getNodeName() !== null) {
+                    $this->requireNodeNameToBeUncovered(
```

Comment:
Do we have other places where we call this method? If so, maybe we should move the check in there and rename the method?

Reply:
No, this was the only place.
But in the meantime there are more methods that expect a non-nullable `NodeName` (see https://github.com/neos/contentrepository-development-collection/blob/master/Neos.EventSourcedContentRepository/Classes/Domain/Context/NodeAggregate/Feature/NodeRetyping.php#L67-L71)

I think a clean solution would get rid of some nullable types here and introduce some kind of (```NodeName::isUnnamed()```) concept for example

Reply:
I made the `NodeName` parameter nullable for now and moved the decision logic into the constraint check.
I also tweaked some more cases where the node name wasn't nullable yet.

Reply:
The above solution is my preferred one, we should imho just in addition introduce NodeAggregate::isNamed to make this more readable. "If (the) node aggregate is named, require (this) node name to be uncovered) is a complete sentence that makes total sense at this point

Reply:
@nezaniel can you clarify what you meant by "above solution"? Is this PR fine in your POV? (it currently blocks the routing change)